classes/Search.php

```diff
@@ -326,11 +326,12 @@ public static function find(
 					p.`id_product` = pl.`id_product`
 					AND pl.`id_lang` = '.(int)$id_lang.Shop::addSqlRestrictionOnLang('pl').'
 				)
-				'.(Combination::isFeatureActive() ? 'LEFT JOIN `'._DB_PREFIX_.'product_attribute_shop` product_attribute_shop
-				ON (p.`id_product` = product_attribute_shop.`id_product` AND product_attribute_shop.`default_on` = 1 AND product_attribute_shop.id_shop='.(int)$context->shop->id.')':'').'
+				'.(Combination::isFeatureActive() ? 'LEFT JOIN `'._DB_PREFIX_.'product_attribute_shop` product_attribute_shop FORCE INDEX (id_product)
+				    ON (p.`id_product` = product_attribute_shop.`id_product` AND product_attribute_shop.`default_on` = 1 AND product_attribute_shop.id_shop='.(int)$context->shop->id.')':'').'
 				'.Product::sqlStock('p', 0).'
-				LEFT JOIN `'._DB_PREFIX_.'manufacturer` m ON m.`id_manufacturer` = p.`id_manufacturer`
-				LEFT JOIN `'._DB_PREFIX_.'image_shop` image_shop
+				LEFT JOIN `'._DB_PREFIX_.'manufacturer` m FORCE INDEX (PRIMARY) 
+				    ON m.`id_manufacturer` = p.`id_manufacturer`
+				LEFT JOIN `'._DB_PREFIX_.'image_shop` image_shop FORCE INDEX (id_product)
```

Comment:
These two others `FORCE INDEX` weren't obvious enough for MySQL?

Reply:
I saw 1 cases where MySQL was also messing up with manufacturer. Since it speeds up a bit the computing of the execution plan, there's no harm adding those extra FORCE INDEX :)

Reply:
Yep, but I had to ask :)